### ISSUE TITLE
Versus vim-smartchr

### ISSUE THREAD

**Issue by kana** [kana/vim-smartinput#2]
For example,
- `IEnumerable#` -- `<` -> `IEnumerable<#>` (with [vim-smartpunc](https://github.com/kana/vim-smartpunc)) is useful to write type pameters in C#.
- `_<_` <--> `<` (read `_` as a whitespace / with [vim-smartchr](https://github.com/kana/vim-smartchr)) is useful to write comparison expressions.

Both configurations are conflicted.  What is the best way to coexist such configurations?
Some thoughts:
- Both plugins have the similar purpose that is to support user's input.
- vim-smartpunc is mostly a superset of vim-smartchr.
  So that conflicts can be resolved by using only vim-smartpunc
  if it supports features which are available only in vim-smartchr.
- vim-smartchr is available in both Insert mode and Command-line mode,
  while vim-smartpunc is available only in Insert mode.
  - It's hard to make vim-smartpunc available in Command-line mode
    because of the current implementation,
    especially for interpretation of `"at"` in Command-line mode.
  - But Command-line mode is the mode to input Ex commands interactively.
    Though input assistants are useful to write source code,
    I doubt that such assistants are also useful in Command-line mode,
    because most input is very short commands such as `:wqall`
    or commands with special syntax such as `:s/.../.../g`.
    It's rarely to type complex expressions in Command-line mode.
    So, is it really necessary to support Command-line mode?
- vim-smartpunc provides flexible rules to configure its behavior,
  while vim-smarchr is not so flexible.
  - On vim-smartpunc, it's easy to write context-sensitive rules with `"at"`, `"filetype"` and `"syntax"`.
  - On vim-smartchr,
    `"at"` is not configurable (`'foo\%#'`-style only),
    `"filetype"` depends on how rules are defined (e.g. via filetype plugins or `:autocmd FileType`)
    and `"syntax"` is not supported completely.
  - On vim-smartpunc, it's possible to specify an arbitrary `"input"` for any rule.
  - On vim-smartchr,
    `"input"` is limited to a normal string,
    in other words, it's not possible to adjust the cursor position.
